utils/python/CIME/SystemTests/system_tests_compare_two.py

```diff
@@ -0,0 +1,357 @@
+                # case, but if we didn't remove the case2 directory, the next
+                # re-build of the test would think, "okay, setup is done, I can
+                # move on to the build", which would be wrong.
+                shutil.rmtree(self._caseroot2)
+                self._activate_case1()
+                logger.warning("WARNING: Test case setup failed. Case2 has been removed, "
+                               "but the main case may be in an inconsistent state. "
+                               "If you want to rerun this test, you should create "
+                               "a new test rather than trying to rerun this one.")
+                raise
+
+    def _case_from_existing_caseroot(self, caseroot):
+        """
+        Returns a Case object from an existing caseroot directory
+
+        Args:
+            caseroot (str): path to existing caseroot
+        """
+        return Case(case_root=caseroot, read_only=False)
+
+    def _activate_case1(self):
+        """
+        Make case 1 active for upcoming calls
+        """
+        os.chdir(self._caseroot1)
+        self._set_active_case(self._case1)
+
+    def _activate_case2(self):
+        """
+        Make case 2 active for upcoming calls
+        """
+        os.chdir(self._caseroot2)
+        self._set_active_case(self._case2)
+
+    def _setup_cases(self):
+        """
+        Does all test-specific set up for the two test cases.
+        """
+
+        # Set up case 1
+        self._activate_case1()
+        self._common_setup()
+        self._case_one_setup()
+        # Flush the case so that, if errors occur later, then at least case 1 is
+        # in a correct, post-setup state. This is important because the mere
+        # existence of a case 2 directory signals that setup is done. So if the
+        # build fails and the user rebuilds, setup won't be redone - so it's
+        # important to ensure that the results of setup are flushed to disk.
+        #
+        # Note that case 1 will be in its post-setup state even if case 2 setup
+        # fails. Putting the case1 flush after case 2 setup doesn't seem to help
+        # with that (presumably some flush is called automatically), and anyway
+        # wouldn't help with things like appending to user_nl files (which don't
+        # rely on flush). So we just have to live with that possibility (but
+        # note that we print a warning to the log file if that happens, in the
+        # caller of this method).
+        self._case.flush()
+
+        # Set up case 2
+        self._activate_case2()
+        self._common_setup()
+        self._case_two_setup()
+        # Flush the case so that, if errors occur later, then at least case2 is
+        # in a correct, post-setup state
+        self._case.flush()
+
+        # Go back to case 1 to ensure that's where we are for any following code
+        self._activate_case1()
+
+    def _force_case2_settings(self):
```

Comment:
Why not use case.submit for case2?   Are you running case2 in the same submit as case 1 and thus using the same environment and pe-layout?  That seems rather over restrictive since environment and pe-layout might be among the things that we want to test.